src/server_manager/web_app/digitalocean_app/create_server_app.ts

```diff
@@ -0,0 +1,271 @@
+  sgp: `${FLAG_IMAGE_DIR}/singapore.png`,
+  blr: `${FLAG_IMAGE_DIR}/india.png`,
+  fra: `${FLAG_IMAGE_DIR}/germany.png`,
+  lon: `${FLAG_IMAGE_DIR}/uk.png`,
+  sfo: `${FLAG_IMAGE_DIR}/us.png`,
+  tor: `${FLAG_IMAGE_DIR}/canada.png`,
+  nyc: `${FLAG_IMAGE_DIR}/us.png`,
+};
+
+/**
+ * A web component that guides a user through the process of creating an Outline
+ * server on DigitalOcean.
+ *
+ * Applications must call {@link start} to begin the server creation flow. The flow
+ * is asynchronous and events will be fired on successful server creation or user
+ * cancellation.
+ */
+@customElement('digitalocean-create-server-app')
+export class DigitalOceanCreateServerApp extends LitElement {
+  /**
+   * Event fired upon successful completion of the DigitalOcean create server flow.
+   * Note that even though the event contains the newly created server, the server
+   * may still be in the process of initializing.
+   *
+   * @event DigitalOceanCreateServerApp#server-created
+   * @property {ManagedServer} server - The newly created ManagedServer domain model.
+   */
+  public static EVENT_SERVER_CREATED = 'server-created';
+
+  /**
+   * Event fired when the user cancels the create server flow.
+   *
+   * @event DigitalOceanCreateServerApp#server-create-cancelled
+   */
+  public static EVENT_SERVER_CREATE_CANCELLED = 'server-create-cancelled';
+
+  @property({type: Function}) localize: Function;
+  @property({type: Object}) notificationManager: OutlineNotificationManager;
+
+  private currentPage = 'loading';
+  private regionPicker: OutlineRegionPicker;
+  private serverRepository: DigitaloceanServerRepository;
+
+  static get styles() {
+    return [
+      COMMON_STYLES, css`
+      .container {
+        display: flex;
+        flex-direction: column;
+        justify-content: center;
+        height: 100%;
+        align-items: center;
+        padding: 132px 0;
+        font-size: 14px;
+      }
+      .card {
+        display: flex;
+        flex-direction: column;
+        align-items: stretch;
+        justify-content: space-between;
+        margin: 24px 0;
+        padding: 24px;
+        background: var(--background-contrast-color);
+        box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.14), 0 2px 2px 0 rgba(0, 0, 0, 0.12), 0 1px 3px 0 rgba(0, 0, 0, 0.2);
+        border-radius: 2px;
+      }
+      @media (min-width: 1025px) {
+        paper-card {
+          /* Set min with for the paper-card to grow responsively. */
+          min-width: 600px;
+        }
+      }
+      .card p {
+        color: var(--light-gray);
+        width: 100%;
+        text-align: center;
+      }
+      .card paper-button {
+        color: var(--light-gray);
+        width: 100%;
+        border: 1px solid rgba(255, 255, 255, 0.12);
+        border-radius: 2px;
+      }
+      .card paper-button[disabled] {
+        color: var(--medium-gray);
+        background: transparent;
+      }
+      /* Mirror images */
+      :host(:dir(rtl)) .mirror {
```

Comment:
Perhaps you can reproduce a bit of the [DirMixin logic](https://github.com/Polymer/polymer/blob/master/lib/mixins/dir-mixin.js)

What you can do i set the host `dir` attribute on connected, based on the `<html>` `dir` attribute, if the host attribute is not already set. See https://github.com/Polymer/polymer/blob/master/lib/mixins/dir-mixin.js#L176

Ideally it would be inherited, but I don't know how to do that.

Once you have the `dir` attribute, you can do
```css
:host([dir="rtl"]) .mirror {
}
```

Reply:
Perhaps you can use this to calculate the inherited direction:
```js
this.setAttribute("dir", getComputedStyle(this).direction())
```

Reply:
cc: @JonathanDCohen

Reply:
The gallery app may be a great place to play with that.